packages/core/src/timeline.ts

```diff
@@ -76,6 +77,15 @@ export class Timeline {
       }
     }
 
+    Inspector.trace({
+      timestamp: new Date().toISOString(),
+      stage: 'delivered',
+      event: result as never,
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      id: incomingEvent.messageId!,
```

Comment:
nit: I don't think this should ever ever happen, but maybe use `result` instead of `incomingEvent` as technically a plugin can change the `messageId`